vermouth/tests/test_map_input.py

```diff
@@ -226,21 +237,97 @@ def test_read_mapping(case):
 2 X1 Y U
     """,  # Multiple difinitions for the same atom
     """
-no initial context
-    """,
-    """
-[ molecule ]
 [ atoms ]
 0 A B
     """,  # no molecule name
+    """
+Just a pile of garbage.
+Clearly, this is not a mapping file.
+Not even a partial one.
+    """,
+    # If a file contains two consecutive [ molecule ] lines, it means the first
+    # of the molecules is empty.
+    """
+[ molecule ]
+    """,
 ))
-
 def test_read_mapping_errors(content):
     """
     Test that syntax error are caught when reading a mapping.
     """
     with pytest.raises(IOError):
-        vermouth.map_input.read_mapping(content.split('\n'))
+        vermouth.map_input._read_mapping_partial(content.split('\n'), 1)  # pylint: disable=protected-access
```

Comment:
Move the pylint stanza to the top of the file. Otherwise I expect it'll be all over.